### ISSUE TITLE
When Local Clock set to 12AM hour, Chrome returns incorrect TZ values

### ISSUE THREAD

**Issue by beatzball** [iamkun/dayjs#1035]
**Describe the bug**
Steps:
1. set your local clock time on your machine to the 12AM hour
2. have a script that loads dayjs and extends utc & timezone
3. open firefox and run command in console:
4. open chrome or MS edge and run the same command in console
    ```
    dayjs
      .tz(
        dayjs().format('YYYY-MM-DD HH:mm'), // get local date and time
        'America/Los_Angeles'
      )
      .day(2)                               // set it to Tuesday
      .hour(23)                             // set the hour late in the day
      .minute(1)                            // set some random minute
      .tz('America/Chicago')                // now convert it to a tz for the next day (should be wednesday)
      .format(`dddd h:mm A [America/Chicago]`);
    ```

**Expected behavior**
The day should be set to Wednesday when converting, but Chrome is saying it's Tuesday

![Developer_Tools_-_Tuesday_2_52_AM_America_Chicago_-_http___localhost_7177_time-converter_day_2_month_8_hour_23_minute_28_tz_93_otz_51_and_DevTools_-_localhost_7177_time-converter_day_2_month_8_hour_23_minute_28_tz_93_otz_51_and_Date___Time_](https://user-images.githubusercontent.com/38116726/92456258-8ac83400-f177-11ea-8842-df30d3eacfdf.jpg)

**Information**
 - Day.js Version: 1.8.35
 - OS: MacOS Catalina 10.15.6
 - Browser: 
    - Failing Chrome Based browsers:
        - Chrome 85.0.4183.83
        - MS Edge 85.0.564.44
    - Succeeds in Firefox:
        - Firefox 80.0.1
 - Time zone: GMT-07:00 DST (Pacific Daylight Time)

**Comment by iamkun** [comment]
Interesting. But I'm afraid I can't reproduce this issue.

![image](https://user-images.githubusercontent.com/17680888/92467918-eb518280-f204-11ea-8892-06cb4ab79e67.png)

**Comment by beatzball** [comment]
thanks for the quick reply. 
🧐 based on your screenshot, it looks like it’s in PM, which does still work...it seems to only be 12 AM where it does not.
did you try unchecking the “Set date and time automatically” and setting it to “AM” instead of “PM”?

**Comment by iamkun** [comment]
I've double-checked the setting, according to the system, 12:59 AM(59 minutes passed noon) is exactly the time we want, and there's no time at 12:59 PM (the last one is 11:59 PM)

**Comment by beatzball** [comment]
hmm, i'm a little confused by the `no time at 12:59 PM (the last one is 11:59 PM)`

but for more insight on the former part of your comment...

59 minutes past noon (PM) works for me
![Tuesday_1_28_AM_America_Chicago_and_Date___Time_and_When_Local_Clock_set_to_12AM_hour__Chrome_returns_incorrect_TZ_values_·_Issue__1035_·_iamkun_dayjs](https://user-images.githubusercontent.com/38116726/92508570-b5d57680-f1bd-11ea-8bff-2318e14e1bbb.jpg)


but 59 minutes past midnight (AM) does not
![Tuesday_1_28_AM_America_Chicago_and_When_Local_Clock_set_to_12AM_hour__Chrome_returns_incorrect_TZ_values_·_Issue__1035_·_iamkun_dayjs](https://user-images.githubusercontent.com/38116726/92508484-92aac700-f1bd-11ea-8538-2882ccaa4838.jpg)

**Comment by beatzball** [comment]
although i still think this is a bug, my workaround is that i've explicitly defined the time in the beginning to always be `12:00` instead of relying on the local time (which could at some point be between `00:00` and `00:59`)

```javascript
dayjs
  .tz(
    dayjs().format('YYYY-MM-DD [12:00]'), // get local date and force time of 12:00
    'America/Los_Angeles'
  )
  .day(2)                               // set it to Tuesday
  .hour(23)                             // set the hour late in the day
  .minute(1)                            // set some random minute
  .tz('America/Chicago')                // now convert it to a tz for the next day (should be wednesday)
  .format(`dddd h:mm A [America/Chicago]`);
```

**Comment by iamkun** [comment]
Yes, I can reproduce this issue after changing my system timezone to `GMT-07:00` and time to 12:59 AM (yes, it's available in this locale)

However, I don't know why there's a different result between chrome and firefox.

I'll mark this as a bug and digging it deeply later.
